homeassistant/bootstrap.py

```
@@ -748,111 +753,73 @@ async def _async_resolve_domains_to_setup(
+        for domain in integrations_to_setup
+        for dep in integrations_dependencies[domain].difference(
+            all_integrations_to_setup
+        )
+    )
 
-        if unseen_deps := manifest_deps - integration_cache.keys():
-            # If there are dependencies, try to preload all
-            # the integrations manifest at once and add them
-            # to the list of requirements we need to install
-            # so we can try to check if they are already installed
-            # in a single call below which avoids each integration
-            # having to wait for the lock to do it individually
-            deps = await loader.async_get_integrations(hass, unseen_deps)
-            for dependant_domain, dependant_itg in deps.items():
-                if isinstance(dependant_itg, loader.Integration):
-                    integration_cache[dependant_domain] = dependant_itg
-                    needed_requirements.update(dependant_itg.requirements)
-
-        if resolve_dependencies_tasks:
-            await asyncio.gather(*resolve_dependencies_tasks)
-
-        for itg in integrations_to_process:
-            try:
-                all_deps = itg.all_dependencies
-            except RuntimeError:
-                # Integration.all_dependencies raises RuntimeError if
-                # dependencies could not be resolved
-                continue
-            for dep in all_deps:
-                if dep in domains_to_setup:
-                    continue
-                domains_to_setup.add(dep)
-                to_resolve.add(dep)
-
-    _LOGGER.info("Domains to be set up: %s", domains_to_setup)
+    integrations_requirements = {
+        domain: itg.requirements for domain, itg in integrations_to_process.items()
+    }
+    integrations_after_dependencies = (
+        await loader.resolve_integrations_after_dependencies(
+            hass, integrations_to_process.values(), ignore_exceptions=True
+        )
```

I don't see where the exceptions would ever get raised or logged from.  Can we have this return the exceptions instead and log them so we don't hide CircularDependency ?

---

Here we just collect them all to get their requirements and start loading those early.

We detect and log errors below: https://github.com/home-assistant/core/blob/60b78eddfac7c58d398068c5c40a5b2fc1a77004/homeassistant/bootstrap.py#L914-L919

---

Would you please add a comment about why its safe to ignore the exceptions. Something along the lines that this is only for early loading and any exceptions will be handled in `_async_set_up_integrations`